src/test/PolygonStateBridge.t.sol

```diff
@@ -0,0 +1,196 @@
+        mainnetFork = vm.createFork(MAINNET_RPC_URL);
+
+        vm.selectFork(mainnetFork);
+        /// @notice Roll the fork to a block where FXPortal already has been deployed
+        /// @notice and the Base crossDomainMessenger ResolvedDelegateProxy target address is initialized
+        vm.rollFork(17711915);
+
+        sampleRoot = uint256(0x123);
+        mockWorldID = new WorldIDIdentityManagerMock(sampleRoot);
+        mockWorldIDAddress = address(mockWorldID);
+
+        checkpointManager = address(0x86E4Dc95c7FBdBf52e33D563BbDB00823894C287);
+        fxRoot = address(0xfe5e5D361b2ad62c541bAb87C45a0B9B018389a2);
+
+        polygonStateBridge = new PolygonStateBridge (
+            checkpointManager,
+            fxRoot,
+            mockWorldIDAddress
+        );
+
+        owner = polygonStateBridge.owner();
+    }
+
+    ///////////////////////////////////////////////////////////////////
+    ///                           SUCCEEDS                          ///
+    ///////////////////////////////////////////////////////////////////
+
+    /// @notice select a specific fork
+    function test_canSelectFork_succeeds() public {
+        // select the fork
+        vm.selectFork(mainnetFork);
+        assertEq(vm.activeFork(), mainnetFork);
+    }
+
+    /// @notice tests that a root can be sent successfully to Polygon
+    function test_propagateRoot_succeeds(address randomAddress) public {
+        vm.assume(randomAddress != address(0) && randomAddress != owner);
+
+        vm.expectEmit(true, true, true, true);
+
+        emit RootPropagated(sampleRoot);
+
+        vm.prank(randomAddress);
+        polygonStateBridge.propagateRoot();
+    }
+
+    /// @notice Tests that the owner of the StateBridge contract can transfer ownership
+    /// using Ownable2Step transferOwnership
+    /// @param newOwner the new owner of the contract
+    function test_owner_transferOwnership_succeeds(address newOwner) public {
+        vm.assume(newOwner != address(0));
+
+        vm.expectEmit(true, true, true, true);
+
+        // OpenZeppelin Ownable2Step transferOwnershipStarted event
+        emit OwnershipTransferStarted(owner, newOwner);
+
+        vm.prank(owner);
+        polygonStateBridge.transferOwnership(newOwner);
+
+        vm.expectEmit(true, true, true, true);
+
+        // OpenZeppelin Ownable2Step transferOwnership event
+        emit OwnershipTransferred(owner, newOwner);
+
+        vm.prank(newOwner);
+        polygonStateBridge.acceptOwnership();
+
+        assertEq(polygonStateBridge.owner(), newOwner);
+    }
+
+    /// @notice tests whether the StateBridge contract can set root history expiry on Optimism and Polygon
+    /// @param _rootHistoryExpiry The new root history expiry for OpWorldID and PolygonWorldID
+    function test_owner_setRootHistoryExpiryPolygon_succeeds(uint256 _rootHistoryExpiry) public {
+        vm.assume(_rootHistoryExpiry != 0 && owner != address(0));
```

Comment:
Right now, we are checking that `_rootHistoryExpiry != 0`, but it looks like it is possible to set the expiry to 0. If this is not the intended behavior, we should add a check that `expiry != 0` within the function to set the expiry.

 Let me know if I am missing something.

```js
    function _setRootHistoryExpiry(uint256 expiryTime) internal virtual {
        ROOT_HISTORY_EXPIRY = expiryTime;

        emit RootHistoryExpirySet(expiryTime);
    }
```

Reply:
removed the assumption, I don't think that the logic needs to be baked into setRootHistoryExpiry. If we ever want the expiry to be immediate, it should be an option imo, not a flaw. Thanks for catching it!